06week/Checkpoint2.js

```diff
@@ -0,0 +1,105 @@
+
+
+// Find the sum of the even values
+const sumEvens = strNums.reduce((total, num) =>{
+if(nums % 2==0){
+  return total + num;
+})// Use reduce to get a single value from the array if the nums are even.
+
+console.log(sumEvens);
+
+// Find the index of the first "Austin" value (the value plus its index equals 512)
+const atxIdx = nums.find((item, index))=>{
+  item + index == 512;
+}
+// We will need to use find to get the first occurence.
+console.log(`index: ${atxIdx}, value: ${nums[atxIdx]}`);
+
+const weather = [
+  { id: 5743823523151872,
+    weather_state_name: "Light Cloud",
+    weather_state_abbr: "lc",
+    wind_direction_compass: "NNE",
+    created: "2018-07-11T20:53:03.251710Z",
+    applicable_date: "2018-07-11",
+    min_temp: 14.43,
+    max_temp: 23.36,
+    the_temp: 22.785,
+    wind_speed: 5.682503989556987,
+    wind_direction: 21.6264939172659,
+    air_pressure: 1024.45,
+    humidity: 58,
+    visibility: 8.683041040324504,
+    predictability: 70
+  },
+  { id: 6188149969518592,
+    weather_state_name: "Heavy Cloud",
+    weather_state_abbr: "hc",
+    wind_direction_compass: "NE",
+    created: "2018-07-11T20:53:03.268190Z",
+    applicable_date: "2018-07-12",
+    min_temp: 14.81,
+    max_temp: 25.52,
+    the_temp: 24.61,
+    wind_speed: 3.2461141472739206,
+    wind_direction: 42.72552812997726,
+    air_pressure: 1024.605,
+    humidity: 54,
+    visibility: 10.633835898353615,
+    predictability: 71
+  },
+  { id: 5742049676492800,
+    weather_state_name: "Showers",
+    weather_state_abbr: "s",
+    wind_direction_compass: "E",
+    created: "2018-07-11T20:53:03.947390Z",
+    applicable_date: "2018-07-13",
+    min_temp: 15.5525,
+    max_temp: 25.3475,
+    the_temp: 24.175,
+    wind_speed: 3.6572546846814604,
+    wind_direction: 90.32910675612557,
+    air_pressure: 1025.385,
+    humidity: 57,
+    visibility: 10.181166984808717,
+    predictability: 73
+  },
+  { id: 6696130918219776,
+    weather_state_name: "Heavy Cloud",
+    weather_state_abbr: "hc",
+    wind_direction_compass: "SSW",
+    created: "2018-07-11T20:53:04.068570Z",
+    applicable_date: "2018-07-14",
+    min_temp: 15.915,
+    max_temp: 27.0925,
+    the_temp: 26.585,
+    wind_speed: 3.649847972759087,
+    wind_direction: 200.04283406736377,
+    air_pressure: 1024.4450000000002,
+    humidity: 52,
+    visibility: 11.14056410562316,
+    predictability: 71
+  },
+],
+
+//using a higher order function, create an array of the unique 'weather_state_name' values of the weather array. Your function should return the following array ['Light Cloud', 'Heavy Cloud', 'Showers']
+const weatherStates = weather.forEach((weather_state_name))=>{
+  return weather.weather_state_name;
+}
+//forEach() will log a new array for just weather_state_name.
+console.log(weatherStates)
+//find the id of the object in weather that has a min_temp of 15.915
+
+const idealTemp = weather.find((min_temp))=>{
```

Comment:
don't close your callback too soon.  Also, you would be looping the entire array of objects.  For every object, you would then grab the min_temp.  Let's say you passed obj as your callback param.  you would return obj.min_tem === 15.915